go/v1beta1/storage/storage_test.go

```diff
@@ -0,0 +1,886 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/grafeas/grafeas/go/name"
+	"github.com/grafeas/grafeas/go/v1beta1/api"
+	"github.com/grafeas/grafeas/go/v1beta1/project"
+	pb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	prpb "github.com/grafeas/grafeas/proto/v1beta1/project_go_proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Tests implementations of grafeas.Storage and project.Storage
+// createStore is a function that creates new grafeas.Storag and project.Storage instances and
+// a corresponding cleanUp function that will be run at the end of each
+// test case.
+func doTestStorage(t *testing.T, createStore func(t *testing.T) (grafeas.Storage, project.Storage, func())) {
+	t.Run("CreateProject", func(t *testing.T) {
+		_, gp, cleanUp := createStore(t)
+		defer cleanUp()
+
+		ctx := context.Background()
+		pID := "project"
+		p := &prpb.Project{}
+		p.Name = name.FormatProject(pID)
+		if _, err := gp.CreateProject(ctx, pID, p); err != nil {
+			t.Errorf("CreateProject got %v want success", err)
+		}
+		// Try to insert the same project twice, expect failure.
```

Comment:
Errors may be a little cleaner if you layer a second t.Run for the CreateProject case and the createproject twice case

Reply:
Good idea, added for Create{Project,Note,Occurrence}.